rdr_service/resource/generators/participant.py

```diff
@@ -389,6 +389,30 @@ def _prep_participant(self, p_id, ro_session):
             'date_of_death': deceased_date_of_death
         }
 
+        # Collect participant pairing history
+        pairing = ro_session.query(ParticipantHistory.lastModified, ParticipantHistory.hpoId,
+                                   ParticipantHistory.organizationId, ParticipantHistory.siteId). \
+            filter(ParticipantHistory.participantId == p_id).order_by(ParticipantHistory.lastModified).all()
+
+        pairing_history = None
+        if pairing:
+            pairing_history = list()
+            for item in pairing:
+                hpo = ro_session.query(HPO.name).filter(HPO.hpoId == item.hpoId).first()
```

Comment:
A few things:
Can we pull in the hpo / org ( / site ) details as outer joins on the `pairing` query above so we only have to hit the DB once?

Do we want the organization display string in this record or is the id sufficient?

And, do we also need to get the site details for each history record rather than using what's currently  in the `participant `table record?

Reply:
I can fix up the pairing history query to join the hpo, organization and site tables to get the descriptions, instead of calling for them in the loop.  The site ids will be different across the pairing history, so using the participant summary values doesn't make sense here.